src/http/modules/ngx_http_ssi_filter_module.c

```diff
@@ -366,6 +355,32 @@ ngx_http_ssi_header_filter(ngx_http_request_t *r)
     ngx_str_set(&ctx->errmsg,
                 "[an error occurred while processing the directive]");
 
+    return ctx;
+}
+
+
+static ngx_int_t
+ngx_http_ssi_header_filter(ngx_http_request_t *r)
+{
+    ngx_http_ssi_ctx_t       *ctx, *mctx;
+    ngx_http_ssi_loc_conf_t  *slcf;
+
+    slcf = ngx_http_get_module_loc_conf(r, ngx_http_ssi_filter_module);
+
+    if (!slcf->enable
+        || r->headers_out.content_length_n == 0
+        || ngx_http_test_content_type(r, &slcf->types) == NULL)
+    {
+        return ngx_http_next_header_filter(r);
+    }
+
+    mctx = ngx_http_get_module_ctx(r->main, ngx_http_ssi_filter_module);
+
+    ctx = ngx_http_ssi_create_ctx(r);
```

Comment:
I don't know when `ngx_http_ssi_header_filter` is invoked and if this will result in the context being created again. Should we protect against it or is there a single place that we can create the context that would proceed both stub output and the ssi header filter?